### DISCUSSION TITLE
如何动态（运行时）完全关闭默认的layout布局

### DISCUSSION THREAD

**Discussion by ahuigo** [umijs/umi#10682] [Q&A] [answered]
已经在umijs　在.umirc.ts　中指定了 layout.

可动态关闭layout吗？

需求场景：作为内部页面内嵌到其它主页面（此时，不需要layout菜单）

**Comment by ahuigo** [comment]
勉强用这个对付一下：
```
  api.addHTMLScripts(() => [
    `
    function hideLayout(){
      const sider1 = document.querySelector('.ant-layout>.ant-layout-sider');
      const sider0 = sider1?.previousSibling
      if(sider1 && sider0) {
        sider1.style.display = 'none';
        sider0.style.display = 'none';
        return true
      }
    }
    function isInnerPage () {
      try { return window.self !== window.top; } catch (e) { return true; }
    }
    addEventListener("load", (event) => {
      if(isInnerPage()) {
        hideLayout()
        // setTimeout(hideLayout, 1000)
      }
    });
    `
  ]);
```

**Comment by fz6m** [comment, marked as answer]
在 umi 4 中，通过 `layout: false` 关闭，See https://umijs.org/docs/guides/routes#%E5%85%A8%E5%B1%80-layout

**Reply by ahuigo** [reply]
> 在 umi 4 中，通过 `layout: false` 关闭，See https://umijs.org/docs/guides/routes#%E5%85%A8%E5%B1%80-layout

谢谢，不过：
1.　`src/layouts/index.tsx`, 这个layout是inner layout，umi不能关闭build时由.umirc所开启的外层layout。
2. 　.umirc.ts routes 中的`layouts:false`是build时的，不能运行时再动态修改

可能只能关闭.umirc所开启的外层layout，自己在运行时手动管理layout吧

**Reply by fz6m** [reply]
只能关闭父级的，和你的路由配置表书写方式有关，自己管理路由的话可以使用 [`wrappers`](https://umijs.org/docs/guides/routes#wrappers) 。

**Comment by 662** [comment]
```tsx
export function patchClientRoutes({ routes }) {
  routes[0].element = <Outlet />;
}
```

试试看
